src/backend/src/services/reimbursement-requests.services.ts

```diff
@@ -80,6 +90,79 @@ export default class ReimbursementRequestService {
     return createdReimbursementRequest;
   }
 
+  /**
+   * Edits the given reimbursement Request
+   *
+   * @param id the id of the reimbursement request we are editing
+   * @param dateOfExpense the updated date of expense
+   * @param vendorId the updated vendor id
+   * @param account the updated account
+   * @param expenseTypeId the updated expense type id
+   * @param totalCost the updated total cost
+   * @param reimbursementProducts the updated reimbursement products
+   * @param saboId the updated saboId
+   * @param receiptPictures the updated receipt pictures
+   * @param submitter the person editing the reimbursement request
+   * @returns the edited reimbursement request
+   */
+  static async editReimbursementRequest(
+    requestId: string,
+    dateOfExpense: Date,
+    vendorId: string,
+    account: Club_Accounts,
+    expenseTypeId: string,
+    totalCost: number,
+    reimbursementProducts: ReimbursementProductCreateArgs[],
+    receiptPictures: string[],
+    submitter: User
+  ): Promise<Reimbursement_Request> {
+    const oldReimbursementRequest = await prisma.reimbursement_Request.findUnique({
+      where: { reimbursementRequestId: requestId },
+      include: {
+        reimbursementProducts: true
+      }
+    });
+
+    if (!oldReimbursementRequest) throw new NotFoundException('Reimbursement Request', requestId);
+    if (oldReimbursementRequest.dateDeleted) throw new DeletedException('Reimbursement Request', requestId);
+    if (oldReimbursementRequest.recepientId !== submitter.userId)
+      throw new AccessDeniedException(
+        'You do not have access to delete this reimbursement request, only the creator can edit a reimbursement request'
+      );
+
+    const vendor = await prisma.vendor.findUnique({
+      where: { vendorId }
+    });
+
+    if (!vendor) throw new NotFoundException('Vendor', vendorId);
+
+    const expenseType = await prisma.expense_Type.findUnique({
+      where: { expenseTypeId }
+    });
+
+    if (!expenseType) throw new NotFoundException('Expense Type', expenseTypeId);
+
+    await updateReimbursementProducts(
+      oldReimbursementRequest.reimbursementProducts,
+      reimbursementProducts,
```

Comment:
we should verify that this isn't an empty array, probably wanna also do the same for create, because we don't want requests with no associated products

Reply:
we do do this with create, just added it here.